src/sci-server/form.go

```diff
@@ -32,14 +33,17 @@ func (r *responder) getForm() (f *form, err error) {
 	f.DuckID = r.req.PostFormValue("duckid")
 	f.Confirm = r.req.PostFormValue("confirm")
 	f.User = r.vars.User
+	f.Role = r.req.PostFormValue("role")
 
 	if f.DuckID == "" {
 		f.errors = append(f.errors, errors.New("duckid must be filled out"))
 	}
 	if f.CRN == "" {
 		f.errors = append(f.errors, errors.New("a course must be chosen"))
 	}
-
+	if f.Role == "" {
```

Comment:
This is semi-out-of-scope, but since you have the validation code already, it's probably worth adding the check in here rather than just checking if the value is blank.  The error message may need a tweak, but otherwise it'll be a pretty simple change, and knocks out another of our mini-tasks.